include/can/core/dispatch.hpp

```diff
@@ -136,4 +136,35 @@ class Dispatcher {
     std::tuple<Listener&...> registered;
 };
 
+/**
+ * A CanMessageBufferListener that will dispatch messages to other
+ * CanMessageBufferListeners
+ * @tparam Listener CanMessageBufferListener objects
+ */
+template <CanMessageBufferListener... Listener>
+class NodeDispatcher {
+  public:
+    using ArbitrationIdTest = bool (*)(uint32_t identifier, uint16_t node_id);
```

Comment:
There's no need to have this method take a `node_id`. 

Ideally, this should be:
`using ArbitrationIdTest = std::function<bool (uint32_t identifier)>;`

What that means is that the argument passed as `test` can be either:
- A callable object (implementing operator() )
```
struct CheckForNodeId {
  NodeId node_id;
  bool operator()(uint32_t arbitration_id) {
    // omitting the check.....
    return node_id_is_in_arbitration_id;
  }
};

/// Constructed like...
CheckForNodeId{.node_id=NodeId::head_left};
```

- A plain old function.
```
bool check_for_node_id(uint32_t) {
  // omitting the check.....
  return node_id_is_in_arbitration_id;
}
```